### PR TITLE
FI-775 Implantable Device profile

### PR DESCRIPTION
**Pull request by yunwwang**
This PR has changes for Bulk Data Implantable Device validation test:
1) Add MUST_SUPORT testing
2) Allow tester specifies comma separated Implantable Device type codes. If the list is empty, Inferno validate all Device resource using US Core Implantable Device profile

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Internal ticket for this PR:
- [x] Internal ticket links to this PR
- [x] Internal ticket is properly labeled (Community/Program)
- [x] Internal ticket has a justification for its Community/Program label
- [x] Code diff has been reviewed for extraneous/missing code
- [x] Tests are included and test edge cases
- [x] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: lib/modules/onc_program/bulk_data_group_validation_sequence.rb [hunk 1]
```diff
@@ -158,6 +158,25 @@ def check_file_request(file, klass, validate_all, lines_to_validate, must_suppor
...
+        # if Device type code is not in predefined type code list, validate using FHIR base profile
+        return nil if resource.resourceType == 'Device' && !predefined_device_type?(resource)
+
+        Inferno::ValidationUtil.guess_profile(resource, version)
+      end
+
+      def predefined_device_type?(resource)
+        return false if resource.nil?
+
+        return true if @instance.bulk_device_types_in_group.nil? || @instance.bulk_device_types_in_group.empty?
```
Comment (arscan) [inline lib/modules/onc_program/bulk_data_group_validation_sequence.rb:171]:
```suggestion
        return true if @instance.bulk_device_types_in_group.blank?
```

**Comment by arscan** [comment]
Do you think it is required to have a separate variable for the device type list for bulk data vs single patinet api?  Seems like we could remove redundant code by just having this require `:device_codes`.  What do you think?

**Comment by arscan** [comment]
When I put garbage into the device type code box I still get 'Successfully validated 2 of 2 resources', and no 'MUST SUPPORT' issues.  Is this intentional?

**Comment by yunwwang** [comment]
I can see the no  MUST_SUPPORT error. Which server did you test?
image

**Comment by yunwwang** [comment]
One problem is the device code text box is on the top of input area
image

**Comment by arscan** [comment]
When I type in "asdfasdfsadfasdfasdf" for the device profile type filter, i would expect that i wouldn't filter out all devices, and therefore i don't validate anything, and therefore i skip.  But it just says that I pass `Successfully validated 2 resource(s).`

**Comment by arscan** [comment]
Was unable to recreate the issue after looking closer w/Yunwei, merging.

**Pull request merged**